Tournament

Niedzwiedz opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/niedzwiedz/montekombat/4)

<!-- Reviewable:end -->

Niedzwiedz committed:
Better application view changes

Niedzwiedz committed:
Tournament model alpha

specs, model methods and validations

Niedzwiedz committed:
CRUD controller for tournament

basic crud actions and tests for tournament controller

Niedzwiedz committed:
Better tournament specs, delete test failing

Niedzwiedz committed:
Tournament authorization and tests. Some fixes.

Add authorization, test authorization for tournaments.

Niedzwiedz committed:
Fix - gemfile, better methods in tournament model

wmaciejak commented:
Reviewed 22 of 29 files at r1.
Review status: 22 of 29 files reviewed at latest revision, 5 unresolved discussions.

---

_[app/models/tournament_team.rb, line 10 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSURAHuAo-p9Igkge-u:-KSURAHuAo-p9Igkge-v:b-914uyc) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/tournament_team.rb#L10)):_

> ``` Ruby
>   private
> 
>   def too_many_players
> ```

Name is very descriptive, but in think about a more general name, cause name of validation should describe what it validate. Something like: `players_count_in_tournament`. The same situation with line 14, but choice of name belongs to you ;)

---

_[app/models/user.rb, line 18 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUU0Z6shNGDXtXPWkf:-KSUU0Z7owRK8plSy1Q5:b-ntx1n6) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/user.rb#L18)):_

> ``` Ruby
>   }
> 
>   def member?(tournament)
> ```

If I good understand you check if a user is member of the specific tournament? Please try optimize this method using only variable tournament instead of Tournament table. Search in teams related to this tournament specific user and then return true or false if they not exists in tournament :)

---

_[app/models/user.rb, line 20 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUVK47OwBINhzGF8MR:-KSUVK47OwBINhzGF8MS:b-ifj1g6) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/user.rb#L20)):_

> ``` Ruby
>   def member?(tournament)
>     Tournament.joins(:teams).each do |tourn|
>       return true if tourn == tournament
> ```

It's perfect place to ternary operator, and please do it :)

---

_[app/views/layouts/application.html.slim, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUVyEgzvu9vj9befE0:-KSUVyEgzvu9vj9befE1:b-3sdlfg) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/views/layouts/application.html.slim#L14)):_

> ``` Slim
>       | ||
>       = link_to "New match", new_match_path
>       - if !logged_in?
> ```

Why not:

``` ruby
   - if logged_in?
        - if current_user.admin?
                   ....
     ...
    - else
     ...
```

In my opinion checking if something is true with else is better way instead of checking true with negation.

---

_[spec/models/user_spec.rb, line 30 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUY6tk6Q3Z-bCHQ-q_:-KSUY6tk6Q3Z-bCHQ-qa:bpggoan) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/spec/models/user_spec.rb#L30)):_

> ``` Ruby
>         user.teams << team
>       end
>       it { expect(user.member?(@tournament)).to eq(true) }
> ```

Please create next user which are not in this tournament, and do the same test with `eq(false)`(please create user with descriptive name ofc :P and divide context for 2 situations):

```
it { expect(user_without_tournament.member?(@tournament)).to eq(false) }
```

Good practice for similar methods is testing more than one situation(or in frankly words - edge cases).

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz commented:
Review status: 22 of 29 files reviewed at latest revision, 5 unresolved discussions.

---

_[app/models/tournament_team.rb, line 10 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSURAHuAo-p9Igkge-u:-KSa5uhDCnaVosUyxHZf:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/tournament_team.rb#L10)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Name is very descriptive, but in think about a more general name, cause name of validation should describe what it validate. Something like: `players_count_in_tournament`. The same situation with line 14, but choice of name belongs to you ;)
> </details>
> Done.

---

_[app/models/user.rb, line 18 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUU0Z6shNGDXtXPWkf:-KSa5vvFBU93hHfJlVew:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/user.rb#L18)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> If I good understand you check if a user is member of the specific tournament? Please try optimize this method using only variable tournament instead of Tournament table. Search in teams related to this tournament specific user and then return true or false if they not exists in tournament :)
> </details>
> Done.

---

_[app/models/user.rb, line 20 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUVK47OwBINhzGF8MR:-KSa5wDHy1mXmvQVf-MC:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/models/user.rb#L20)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> It's perfect place to ternary operator, and please do it :)
> </details>
> Done.

---

_[app/views/layouts/application.html.slim, line 14 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUVyEgzvu9vj9befE0:-KSa5y9NZmduSABbY_y6:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/app/views/layouts/application.html.slim#L14)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Why not:
> 
> ``` ruby
>    - if logged_in?
>         - if current_user.admin?
>                    ....
>      ...
>     - else
>      ...
> ```
> 
> In my opinion checking if something is true with else is better way instead of checking true with negation.
> </details>
> Done.

---

_[spec/models/user_spec.rb, line 30 at r1](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSUY6tk6Q3Z-bCHQ-q_:-KSa5yXUiOm7Ogj3blN4:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/2f4fce2be90d3efb3f4eeb24cce2cfd50e540388/spec/models/user_spec.rb#L30)):_
<details><summary>

<i>Previously, wojtekmaciej (Wojciech Maciejak) wrote…</i></summary>



> Please create next user which are not in this tournament, and do the same test with `eq(false)`(please create user with descriptive name ofc :P and divide context for 2 situations):
> 
> ```
> it { expect(user_without_tournament.member?(@tournament)).to eq(false) }
> ```
> 
> Good practice for similar methods is testing more than one situation(or in frankly words - edge cases).
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fix code review

additional changes for tournament, team, user adding combo

Niedzwiedz committed:
[Work in progress] Users and teams in tournaments

Controller accepts json and saves tournaments, teams and users. I use
each method for that. There is probably better way. Also, I am not using
transactions. Now I try to make proper view with vue, so view will be
easier, also generating and sending json will be simpler than in js.

mskv commented:
Reviewed 1 of 29 files at r1, 16 of 20 files at r2.
Review status: 30 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_[app/assets/javascripts/application.js, line 23 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk292XSshiGmXv4N4F:-KSk292XSshiGmXv4N4G:b-zajh86) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/assets/javascripts/application.js#L23)):_

> ``` JavaScript
> //= require vue-resource
> 
> var teams = new Vue({
> ```

I think this should not be part of this PR. Any Vue experiments should have their own branch.

---

_[app/controllers/tournaments_controller.rb, line 24 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk2W3h7tlLdtzgZaB5:-KSk2W3iv5m6tVk3INQL:b4zjxhi) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L24)):_

> ``` Ruby
> 
>   def create
>     @tournament = Tournament.new(tournament_params)
> ```

You asked about this method. 
One option would be to use `accepts_nested_attributes_for` - you've done it below in the Tournament model. But I like to extract such logic to separate components, since it's a complicated logic that could live in its own abstraction:

I'd create a stateless service for this situation:

```
# app/services/initialize_tournament.rb
class InitializeTournament
  class < self
    def call(creator, tournament_params, teams_params)
      Tournament.new(**tournament_params, teams: initialize_teams(teams_params), creator_id: creator.id)
    end

    private

    def initialize_teams(teams_params)
      teams_params.map do |team_params|
        Team.new(**team_params, team_users: initialize_team_users(team_params[:users]))
      end
    end

    def initialize_team_users(team_users_params)
      team_users_params.map do ...
         ...
      end
    end
  end
end
```

The return value of this service would be an unsaved Tournament instance. And you could call `save` on it like you do below. Let me know if you decide to use it and if it even works.

---

_[app/controllers/tournaments_controller.rb, line 50 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7FMGqX_OlGAmMdtl:-KSk7FMGqX_OlGAmMdtm:b4vmwjj) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L50)):_

> ``` Ruby
> 
>   def update
>     unless @tournament.ended?
> ```

There could be an else case here.

---

_[app/models/team.rb, line 14 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7rGdjwfzyhqj4vSQ:-KSk7rGdjwfzyhqj4vSR:b-12zdc5) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/team.rb#L14)):_

> ``` Ruby
>   # ------------------------------------------------------------------
> 
>   # validates :users, presence: true
> ```

What is the problem here? Why does it not work?

---

_[app/models/tournament.rb, line 38 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSkCOy9Q2hHf0yswDnQ:-KSkCOy9Q2hHf0yswDnR:b14uei3) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/tournament.rb#L38)):_

> ``` Ruby
>     # - lub jakis join
>     # - tabela TournamentUser
>     Team.joins(:users).size
> ```

The tables we have:
tournaments -> tournament_teams -> teams -> team_users -> users
To be honest, I don't know how to write it using ActiveRecord, but maybe something like this? 

```
Users.joins(team_users: { teams: tournament_teams }).where("tournament_teams.tournament_id = ?", [self.id]).count
```

Sorry ,I can't do it off the top of my head, let me know if it does not work!

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mskv commented:
Reviewed 1 of 20 files at r2.
Review status: 31 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mskv commented:
Reviewed 1 of 20 files at r2.
Review status: 32 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz commented:
Review status: 32 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_[app/assets/javascripts/application.js, line 23 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk292XSshiGmXv4N4F:-KSohajj9U5eGQ-nCHvi:blb5q0k) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/assets/javascripts/application.js#L23)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> I think this should not be part of this PR. Any Vue experiments should have their own branch.
> </details>
> So I should delete files in next push or just don't commit any more changes that are vue related?

---

_[app/controllers/tournaments_controller.rb, line 24 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk2W3h7tlLdtzgZaB5:-KSpNXmPyivKRqYQggBa:btzo7jm) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L24)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> You asked about this method. 
> One option would be to use `accepts_nested_attributes_for` - you've done it below in the Tournament model. But I like to extract such logic to separate components, since it's a complicated logic that could live in its own abstraction:
> 
> I'd create a stateless service for this situation:
> 
> ```
> # app/services/initialize_tournament.rb
> class InitializeTournament
>   class < self
>     def call(creator, tournament_params, teams_params)
>       Tournament.new(**tournament_params, teams: initialize_teams(teams_params), creator_id: creator.id)
>     end
> 
>     private
> 
>     def initialize_teams(teams_params)
>       teams_params.map do |team_params|
>         Team.new(**team_params, team_users: initialize_team_users(team_params[:users]))
>       end
>     end
> 
>     def initialize_team_users(team_users_params)
>       team_users_params.map do ...
>          ...
>       end
>     end
>   end
> end
> ```
> 
> The return value of this service would be an unsaved Tournament instance. And you could call `save` on it like you do below. Let me know if you decide to use it and if it even works.
> </details>
> what does *\* do and how/on what it operates? Because something is wrong, it throws TypeError (wrong argument String (expected Symbol)) and I don't know why.

---

_[app/controllers/tournaments_controller.rb, line 50 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7FMGqX_OlGAmMdtl:-KSpTSrlE4Vyz9_5lind:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L50)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> There could be an else case here.
> </details>
> Done.

---

_[app/models/team.rb, line 14 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7rGdjwfzyhqj4vSQ:-KSlNWWCJC8LZN3dU8tM:b-2zvovr) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/team.rb#L14)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> What is the problem here? Why does it not work?
> </details>
> I was implementing json for creating tournaments with teams and I wasn't passing users in them so for testing purposes I commented user presence validation. In next push it will be uncomented and probably working :D

---

_[app/models/tournament.rb, line 38 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSkCOy9Q2hHf0yswDnQ:-KSpWrq5cArmcxj-KDTb:b1p64be) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/tournament.rb#L38)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> The tables we have:
> tournaments -> tournament_teams -> teams -> team_users -> users
> To be honest, I don't know how to write it using ActiveRecord, but maybe something like this? 
> 
> ```
> Users.joins(team_users: { teams: tournament_teams }).where("tournament_teams.tournament_id = ?", [self.id]).count
> ```
> 
> Sorry ,I can't do it off the top of my head, let me know if it does not work!
> </details>
> Hmm... it doesn't work, and I don't know either how to do it properly. But with this whole each team belongs to only one tournament idea. I can remove tournament and make it:
> tournament |has_many|->|belongs_to| teams -> team_users -> users.
> It should make things easier, I think?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mskv commented:
Reviewed 1 of 20 files at r2.
Review status: 33 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_[app/assets/javascripts/application.js, line 23 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk292XSshiGmXv4N4F:-KSpZdMIfQYlDPnqyBku:b-fum9hi) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/assets/javascripts/application.js#L23)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> So I should delete files in next push or just don't commit any more changes that are vue related?
> </details>
> To be honest, looking at the Roadmap, thought you were supposed to introduce Vue in 2 weeks. I also saw "dupa" and thought it was temporary ;p.

---

_[app/controllers/tournaments_controller.rb, line 24 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk2W3h7tlLdtzgZaB5:-KSp_DPJdCeo2fWiXaK0:b-yxjojt) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L24)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> what does *\* do and how/on what it operates? Because something is wrong, it throws TypeError (wrong argument String (expected Symbol)) and I don't know why.
> </details>
> it operates on a hash. so if
> 
> ```
> team_params = { name: "team1", other_param: "otherparam" }
> ```
> 
> then
> 
> ```
> Team.new(**team_params)
> ```
> 
> is equal to
> 
> ```
> Team.new(name: team_params.name, other_param: team_params.other_param)
> ```

---

_[app/controllers/tournaments_controller.rb, line 50 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7FMGqX_OlGAmMdtl:-KSp_rN2ll-cIaDHvZ2m:b-rl4fhw) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L50)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> Done.
> </details>
> I meant `unless @tournament.ended?`. It does not have an `else` case.

---

_[app/models/tournament.rb, line 38 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSkCOy9Q2hHf0yswDnQ:-KSpfSpCdNTMRtQZdOp9:bdtxol6) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/tournament.rb#L38)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> Hmm... it doesn't work, and I don't know either how to do it properly. But with this whole each team belongs to only one tournament idea. I can remove tournament and make it:
> tournament |has_many|->|belongs_to| teams -> team_users -> users.
> It should make things easier, I think?
> </details>
> Ok I've got it:
> 
> ```
> User.joins(teams: :tournament_teams).where("tournament_teams.tournament_id = ?", [tournament.id]).count
> ```
> 
> I've checked this one and this one works^.

```
[9] pry(main)> User.joins(teams: :tournament_teams).where("tournament_teams.tournament_id = ?", [tournament.id]).count
   (0.9ms)  SELECT COUNT(*) FROM "users" INNER JOIN "team_users" ON "team_users"."user_id" = "users"."id" INNER JOIN "teams" ON "teams"."id" = "team_users"."team_id" INNER JOIN "tournament_teams" ON "tournament_teams"."team_id" = "teams"."id" WHERE (tournament_teams.tournament_id = 1)
=> 1
```

Coming back to the question: definitely if you decide to go that route and make teams per-tournament, there is no need for the linking table.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Delete many to many tournament_teams

Now tournament has_many teams, moved some tournament creation
logic to service. Fixed things from last code review.

Niedzwiedz commented:
Review status: 21 of 36 files reviewed at latest revision, 10 unresolved discussions.

---

_[app/assets/javascripts/application.js, line 23 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk292XSshiGmXv4N4F:-KSpgPsyOcbB0LHodMdi:bxnwdkv) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/assets/javascripts/application.js#L23)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> To be honest, looking at the Roadmap, thought you were supposed to introduce Vue in 2 weeks. I also saw "dupa" and thought it was temporary ;p.
> </details>
> I decided on vue now, because from my perspective it looks like less work than writing forms with form_for until week 6 (probably week 9 with current speed :D), then deleting all of them in week 6 and changing to vue. 

---

_[app/controllers/tournaments_controller.rb, line 24 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk2W3h7tlLdtzgZaB5:-KT9QvRZ6NCDt7wuJRsa:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L24)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> it operates on a hash. so if
> 
> ```
> team_params = { name: "team1", other_param: "otherparam" }
> ```
> 
> then
> 
> ```
> Team.new(**team_params)
> ```
> 
> is equal to
> 
> ```
> Team.new(name: team_params.name, other_param: team_params.other_param)
> ```
> 
> </details>
> Done.

---

_[app/controllers/tournaments_controller.rb, line 50 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7FMGqX_OlGAmMdtl:-KT9Qwk_T9euBEAgFaBU:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/controllers/tournaments_controller.rb#L50)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> I meant `unless @tournament.ended?`. It does not have an `else` case.
> </details>
> Done.

---

_[app/models/team.rb, line 14 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7rGdjwfzyhqj4vSQ:-KT9DC5uHAKWMljX7kNJ:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/team.rb#L14)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> I was implementing json for creating tournaments with teams and I wasn't passing users in them so for testing purposes I commented user presence validation. In next push it will be uncomented and probably working :D
> </details>
> Done.

---

_[app/models/tournament.rb, line 38 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSkCOy9Q2hHf0yswDnQ:-KT9D7pY7ioOHGh7TlqQ:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/tournament.rb#L38)):_
<details><summary>

<i>Previously, mskv (Michał Podwórny) wrote…</i></summary>



> Ok I've got it:
> 
> ```
> User.joins(teams: :tournament_teams).where("tournament_teams.tournament_id = ?", [tournament.id]).count
> ```
> 
> I've checked this one and this one works^.
> 
> ```
> [9] pry(main)> User.joins(teams: :tournament_teams).where("tournament_teams.tournament_id = ?", [tournament.id]).count
>    (0.9ms)  SELECT COUNT(*) FROM "users" INNER JOIN "team_users" ON "team_users"."user_id" = "users"."id" INNER JOIN "teams" ON "teams"."id" = "team_users"."team_id" INNER JOIN "tournament_teams" ON "tournament_teams"."team_id" = "teams"."id" WHERE (tournament_teams.tournament_id = 1)
> => 1
> ```
> 
> Coming back to the question: definitely if you decide to go that route and make teams per-tournament, there is no need for the linking table.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mkarbowiak commented:
Reviewed 2 of 29 files at r1, 13 of 16 files at r3.
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

_[app/controllers/teams_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDNvcqum0--F3KRB4z:-KTDNvcrmijg_3XyoX1r:b-5yjkn1) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/teams_controller.rb#L2)):_

> ``` Ruby
> class TeamsController < ApplicationController
>   before_action :set_team, only: [:edit, :update]
> ```

Why this way? I saw something like that in another intern's code, did you took from some tutorial or something?

---

_[app/controllers/tournaments_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDOSdOAs1sG8Yy3LQK:-KTDOSdOAs1sG8Yy3LQL:brhumzt) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/tournaments_controller.rb#L2)):_

> ``` Ruby
> class TournamentsController < ApplicationController
>   before_action :set_tournament, only: [:show, :edit, :update, :check_if_user_is_creator,
> ```

Same here.

---

_[app/controllers/tournaments_controller.rb, line 8 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDOZbNgd2rieLEifxQ:-KTDOZbNgd2rieLEifxR:bq3tz8) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/tournaments_controller.rb#L8)):_

> ``` Ruby
> 
>   before_action :check_if_user_is_creator, only: [:edit, :update, :destroy]
>   def index
> ```

Just a styling issue: please move those before_actions to be one on top another and leave blank line on top of index here.

---

_[app/models/team.rb, line 14 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7rGdjwfzyhqj4vSQ:-KTDPARacIAozhwouQr9:bit4zi1) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/team.rb#L14)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> Done.
> </details>
> Is it done?

---

_[app/models/tournament.rb, line 36 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDPUIzyFEOJ63pRBJO:-KTDPUIzyFEOJ63pRBJP:b-wzt3z7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/models/tournament.rb#L36)):_

> ``` Ruby
> 
>   def number_of_slots
>     (number_of_players_in_team * number_of_teams)
> ```

Redundant parentheses. I really like method names, tough! This code reads really well.

---

_[app/models/tournament.rb, line 42 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDPhCCPNeI_JMmxQ12:-KTDPhCCPNeI_JMmxQ13:bjhih20) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/models/tournament.rb#L42)):_

> ``` Ruby
>     teams.each do |team|
>       players_count_in_team(team)
>     end
> ```

Could fit in one line.

---

_[app/services/initialize_tournament.rb, line 16 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDQmDIe1G9C-q-AkAV:-KTDQmDIe1G9C-q-AkAW:b-r9fene) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L16)):_

> ``` Ruby
> 
>       rescue => ex
>         return ex
> ```

I am not a big fan of those names. We have a plenty of room to name it properly. 

---

_[app/services/initialize_tournament.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRAq0LsPEaiMVdnss:-KTDRAq0LsPEaiMVdnst:b-lob1yv) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L24)):_

> ``` Ruby
>       teams = []
>       teams_params.map do |team_params|
>         team = Team.new(name: team_params[1]["name"], tournament: tournament)
> ```

Why it's always `team_params[1]` or `team_user_params[1]`? How do they look like?

---

_[app/services/initialize_tournament.rb, line 40 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRLZwPFBenQksrz5q:-KTDQwK-6cIu9vrpdprI:bs2xc3) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L40)):_

> ``` Ruby
>       end
>       users
>     end
> ```

Those two methods could be done in more functional way. They are alright, though, so if you don't want to spend more time on that, it's OK. If you're up to a challenge I would suggest looking into Ruby `inject` method.

---

_[config/routes.rb, line 13 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRlKpLiqSeCQSdFbv:-KTDRlKqL8WXMp69i7Ww:b-ctanop) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/config/routes.rb#L13)):_

> ``` Ruby
>   resources :tournaments do
>     collection do
>       get 'types'
> ```

Please always be consistent. You're using double quotes across this file.

---

_[spec/controllers/teams_controller_spec.rb, line 11 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDS-J0EtS0N0fwU8TK:-KTDS-J1Px-7FX_-6kE3:b2n2h01) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L11)):_

> ``` Ruby
>     it "renders the :show view" do
>       expect(response).to render_template("new")
>     end
> ```

There is no need to write 3 lines of test with description:
`it { expect(response).to render_template("new") }`
reads the same as `renders the :show view`.  The same for above.

---

_[spec/controllers/teams_controller_spec.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSRaN4Cf7TYKYv445:-KTDSRaOi3QTqMJfJ6Gh:b-om2t99) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L24)):_

> ``` Ruby
>       it do
>         post_create
>         # Getting weird responses
> ```

What's the weird response? :smile: 

---

_[spec/controllers/tournaments_controller_spec.rb, line 16 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSZWBHChp-LSZ8aoW:-KTDSZWC7bUBCrCseLc8:bcrkdi8) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/tournaments_controller_spec.rb#L16)):_

> ``` Ruby
>     it "renders the :index view" do
>       expect(response).to render_template("index")
>     end
> ```

Same about the descriptions. If your test reads well, like:
`it { expect(response).to render_template("index") }`
and fits into 100 columns, there's no need to write descriptions that provides the same information.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Fixes for #1 Jez code review

fixed some style issues around tournament creation feature

Niedzwiedz commented:
Review status: all files reviewed at latest revision, 22 unresolved discussions.

---

_[app/controllers/teams_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDNvcqum0--F3KRB4z:-KTDTY_SHrkb-jzRff7Z:b-pwps6w) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/teams_controller.rb#L2)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Why this way? I saw something like that in another intern's code, did you took from some tutorial or something?
> </details>
> I think Kamila suggested this. It looks better, especially when we are setting same team in few actions. Also I've seen such practices in many tutorials.

---

_[app/controllers/tournaments_controller.rb, line 8 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDOZbNgd2rieLEifxQ:-KTDUQNXl1VRzKLv-daV:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/tournaments_controller.rb#L8)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Just a styling issue: please move those before_actions to be one on top another and leave blank line on top of index here.
> </details>
> Done.

---

_[app/models/team.rb, line 14 at r2](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KSk7rGdjwfzyhqj4vSQ:-KTDUUhWpvWLhk7et3Ti:b-f29x25) ([raw file](https://github.com/niedzwiedz/montekombat/blob/70463fd4369b2c293c91574494360310843c084a/app/models/team.rb#L14)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Is it done?
> </details>
> yep, in the end I forgot to uncomment :(

---

_[app/models/tournament.rb, line 36 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDPUIzyFEOJ63pRBJO:-KTDVjRpO9Ljwm2x_SZw:b-n21qd0) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/models/tournament.rb#L36)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Redundant parentheses. I really like method names, tough! This code reads really well.
> </details>
> ooooh... Thank you! :3

---

_[app/models/tournament.rb, line 42 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDPhCCPNeI_JMmxQ12:-KTDWC4DXfFj6ej02bfi:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/models/tournament.rb#L42)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Could fit in one line.
> </details>
> Done.

---

_[app/services/initialize_tournament.rb, line 16 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDQmDIe1G9C-q-AkAV:-KTDWlynYQtCIORXUwZ0:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L16)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> I am not a big fan of those names. We have a plenty of room to name it properly. 
> </details>
> Done.

---

_[app/services/initialize_tournament.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRAq0LsPEaiMVdnss:-KTDXEcyAZhHlIBs_q1k:bq126jh) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L24)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Why it's always `team_params[1]` or `team_user_params[1]`? How do they look like?
> </details>
> hmm... Data that I send from view, looks like: {"0"=>{"name"=>"somename", "users"=>{"0"=>{"id"=>"1"}, "1"=>{"id"=>"2"}}}, "1"=>{ "name" ...
> Then after map it gets splited so [0] element is "0" or "1" or
>  ... , and [1] is holding data ( name, users ). I will try to make it better once I finish with editing tournament.

---

_[app/services/initialize_tournament.rb, line 40 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRLZwPFBenQksrz5q:-KTDZ3PK0PFvpl7jSw3h:bwmoibl) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L40)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Those two methods could be done in more functional way. They are alright, though, so if you don't want to spend more time on that, it's OK. If you're up to a challenge I would suggest looking into Ruby `inject` method.
> </details>
> To be honest, I don't like how this code looks, and I want to make it better as soon as possible. But for the next few days I want to finish feature: editing and joining tournament.

---

_[config/routes.rb, line 13 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRlKpLiqSeCQSdFbv:-KTDZsc7jK5tinukUQ4N:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/config/routes.rb#L13)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Please always be consistent. You're using double quotes across this file.
> </details>
> Done.

---

_[spec/controllers/teams_controller_spec.rb, line 11 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDS-J0EtS0N0fwU8TK:-KTDa8fuPrmh84WI7OI_:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L11)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> There is no need to write 3 lines of test with description:
> `it { expect(response).to render_template("new") }`
> reads the same as `renders the :show view`.  The same for above.
> </details>
> Done.

---

_[spec/controllers/teams_controller_spec.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSRaN4Cf7TYKYv445:-KTDaCDIxLD90mSzLaSx:bjekcnk) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L24)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> What's the weird response? :smile: 
> </details>
> I get <200: OK> when I am expecting redirection to team  O.o

---

_[spec/controllers/tournaments_controller_spec.rb, line 16 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSZWBHChp-LSZ8aoW:-KTDekOlykX3Qtk1bilW:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/tournaments_controller_spec.rb#L16)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Same about the descriptions. If your test reads well, like:
> `it { expect(response).to render_template("index") }`
> and fits into 100 columns, there's no need to write descriptions that provides the same information.
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mkarbowiak commented:
Reviewed 8 of 8 files at r4.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

_[app/controllers/teams_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDNvcqum0--F3KRB4z:-KTEM33E0YiUKLnLjGwp:bsstv8) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/teams_controller.rb#L2)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> I think Kamila suggested this. It looks better, especially when we are setting same team in few actions. Also I've seen such practices in many tutorials.
> </details>
> Please use lazy loading.

---

_[app/controllers/teams_controller.rb, line 29 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTEM6Nou1Woz6GfUoIj:-KTEM6NqZ5iXzqGaRz1Y:bay27cb) ([raw file](https://github.com/niedzwiedz/montekombat/blob/9ac1f4775f5142b73d1f0fea29dde5e158225da1/app/controllers/teams_controller.rb#L29)):_

> ``` Ruby
>     if @team.update_attributes(team_params)
>       respond_to do |format|
>         format.html { redirect_to @team, notice: "Team was successfully updated." }
> ```

Is this working? Shouldn't it be `team_path(@team)`?

---

_[app/services/initialize_tournament.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDRAq0LsPEaiMVdnss:-KTEMeyLJ7oPTy_fMWqz:b-oyn09c) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/services/initialize_tournament.rb#L24)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> hmm... Data that I send from view, looks like: {"0"=>{"name"=>"somename", "users"=>{"0"=>{"id"=>"1"}, "1"=>{"id"=>"2"}}}, "1"=>{ "name" ...
> Then after map it gets splited so [0] element is "0" or "1" or
>  ... , and [1] is holding data ( name, users ). I will try to make it better once I finish with editing tournament.
> </details>
> Ok, and next time when posting code here use single tick for one-liners `
> `like that`
> or triple tick ``` for multiline code (you can also provide lang name: ```ruby)
> 
> ```
> like
> that
> ```

---

_[spec/controllers/teams_controller_spec.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSRaN4Cf7TYKYv445:-KTENAmOaXcPKqS7_Djb:bfeq2l6) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L24)):_
<details><summary>

<i>Previously, Niedzwiedz (Michał Nowicki) wrote…</i></summary>



> I get <200: OK> when I am expecting redirection to team  O.o
> </details>
> Maybe it's the issue with that redirecting I mentioned above.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

Niedzwiedz committed:
Lazy loading in controllers

Added lazy loading with controllers. <200:OK> bug still persists.

Niedzwiedz commented:
Review status: 31 of 37 files reviewed at latest revision, 14 unresolved discussions.

---

_[app/controllers/teams_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDNvcqum0--F3KRB4z:-KTEsPlpG4zeGsDoBEuQ:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/teams_controller.rb#L2)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Please use lazy loading.
> </details>
> Done.

---

_[app/controllers/teams_controller.rb, line 29 at r4](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTEM6Nou1Woz6GfUoIj:-KTEa3vejJl9Si9pl4Jy:b-moj7e7) ([raw file](https://github.com/niedzwiedz/montekombat/blob/9ac1f4775f5142b73d1f0fea29dde5e158225da1/app/controllers/teams_controller.rb#L29)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Is this working? Shouldn't it be `team_path(@team)`?
> </details>
> Hmmm... that also was mistake, I don't have `show` team action, so it can't redirect to show team. But if such action existet `@team` should work. (as in: `matches controller`) I am now redirecting to `matches_path` and `root_path`. And both redirections end up as `<200: OK>`

---

_[app/controllers/tournaments_controller.rb, line 2 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDOSdOAs1sG8Yy3LQK:-KTEu3fT3Mb9n7Bj_st3:b-896fix) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/app/controllers/tournaments_controller.rb#L2)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Same here.
> </details>
> Done.

---

_[spec/controllers/teams_controller_spec.rb, line 24 at r3](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTDSRaN4Cf7TYKYv445:-KTEtV_yUHKHlrg_9C3s:bw9b48f) ([raw file](https://github.com/niedzwiedz/montekombat/blob/e8ce1844fb9759cdc8938dad954753f5300a874e/spec/controllers/teams_controller_spec.rb#L24)):_
<details><summary>

<i>Previously, mkarbowiak (Mateusz Karbowiak) wrote…</i></summary>



> Maybe it's the issue with that redirecting I mentioned above.
> </details>
> Definetly something is wrong with `update` and `create` actions . And I only encounter `<200: OK>` bug in `create` and `update` actions of `teams` and after a bit of digging in `create` and `update` actions of `tournaments`. No idea why

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mkarbowiak commented:
Reviewed 3 of 6 files at r5.
Review status: 34 of 37 files reviewed at latest revision, 12 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->

mkarbowiak commented:
Reviewed 3 of 6 files at r5.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

_[spec/controllers/tournaments_controller_spec.rb, line 38 at r5](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4#-KTTV0FZkveRUmgMPkmo:-KTTV0FZkveRUmgMPkmp:b-wyzl02) ([raw file](https://github.com/niedzwiedz/montekombat/blob/32fee751c6d9df0e95c6131bf304991b6af7581e/spec/controllers/tournaments_controller_spec.rb#L38)):_

> ``` Ruby
>   end
>   describe "POST #create" do
>     subject(:post_create) { post :create, params: { tournament: tournament.attributes, teams: team.attributes } }
> ```

Please use `do / end` for too long lines and multilines blocks.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/niedzwiedz/montekombat/4)_

<!-- Sent from Reviewable.io -->
